examples/aerial_stake_optimizer.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2021 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+import json
+
+from cosmpy.aerial.client import LedgerClient
+from cosmpy.aerial.client.distribution import create_withdraw_delegator_reward
+from cosmpy.aerial.client.staking import create_delegate_msg
+from cosmpy.aerial.config import NetworkConfig
+from cosmpy.aerial.tx import SigningCfg, Transaction
+from cosmpy.aerial.wallet import LocalWallet
+from cosmpy.crypto.address import Address
+from cosmpy.crypto.keypairs import PrivateKey
+from cosmpy.protos.cosmos.bank.v1beta1.query_pb2 import QueryTotalSupplyRequest
+from cosmpy.protos.cosmos.params.v1beta1.query_pb2 import QueryParamsRequest
+from cosmpy.protos.cosmos.staking.v1beta1.query_pb2 import QueryValidatorsRequest
+
+
+def M(x, f, S, k, D):
+    return (S * (1 + (k * x)) ** (D / x)) + (
+        (1 - ((1 + (k * x)) ** (D / x))) / (k * x)
+    ) * f
+
+
+def main():
+    ledger = LedgerClient(NetworkConfig.fetchai_dorado_testnet())
+
+    # Set initial stake and desired stake period
+    initial_stake = 50000000000000000000
+    total_period = 60000
+
+    req = QueryValidatorsRequest()
+    resp = ledger.staking.Validators(req)
+
+    # Calculate the total staked in the testnet
+
+    total_stake = 0
+    # validator.status == 3 refers to bonded validators
+    validators_stake = [
+        int(validator.tokens) for validator in resp.validators if validator.status == 3
+    ]
+    total_stake = sum(validators_stake)
+
+    # Choose a validator
```

Comment:
Would it make sense to get the commission rates of all the validators, then choose a random one from those that have the highest commission rate?

Reply:
Yes thats a good idea